### PR TITLE
1901/application program partners

### PR DESCRIPTION
**Pull request by dominikx96**
# Pull Request Template

## Issue Overview

This PR addresses #1901

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds program questions section into the application management form.

## How Can This Be Tested/Reviewed?

1. Create or use existing listing with defined program questions
2. Add a new application and try multiple program answers combination
3. Make sure that data has been saved correctly
4. Try to edit an application
5. Make sure that edit progress saved proper values

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/pages/applications/household/programs.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ const ApplicationPrograms = () => {
   const { conductor, application, listing } = useFormConductor("programs")
   const programs = listing?.listingPrograms
   const uniquePages = new Set(programs?.map((item) => item.ordinal)).size
+  console.info(uniquePages)
```
Comment (YazeedLoonat) [inline sites/public/pages/applications/household/programs.tsx:28]:
```suggestion
```

File: sites/partners/src/applications/PaperApplicationDetails/sections/DetailsPrograms.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+const DetailsPrograms = ({ listingId }: DetailsProgramsProps) => {
+  const { listingDto } = useSingleListingData(listingId)
+
+  const application = useContext(ApplicationContext)
+
+  const listingPrograms = listingDto?.listingPrograms
+  const programs = application?.programs
+
+  const hasMetaData = useMemo(() => {
+    return !!listingPrograms?.filter((listingProgram) => listingProgram.program?.formMetadata)
```
Comment (YazeedLoonat) [inline sites/partners/src/applications/PaperApplicationDetails/sections/DetailsPrograms.tsx:19]: I think its better to use `.some` then `.filter` here

**Review by YazeedLoonat — CHANGES_REQUESTED** [review]
Looks good, just a few code change requests

File: sites/partners/src/applications/PaperApplicationDetails/sections/DetailsPrograms.tsx (hunk 2 above)
Comment (dominikx96) [reply sites/partners/src/applications/PaperApplicationDetails/sections/DetailsPrograms.tsx]: Done, ty!

**Review by jaredcwhite — APPROVED** [review]
LGTM

**Review by YazeedLoonat — APPROVED** [review]
LGTM

**Pull request merged**